pkg/cmd/upgrade/upgrade_cli.go

```diff
@@ -63,9 +62,8 @@ func NewCmdUpgradeCLI(commonOpts *opts.CommonOptions) *cobra.Command {
 
 // Run implements the command
 func (o *UpgradeCLIOptions) Run() error {
-	if config.LoadActiveInstallProfile() == config.CloudBeesProfile {
-		o.NoBrew = true
-	}
+	// upgrading to a specific version is not yet supported in brew so lets disable it for upgrades
+	o.NoBrew = true
```

Comment:
This I think is a problem as you can end up with a brew and non-brew install, which has the potential to screw up your path. Looks like brew does support versions https://stackoverflow.com/questions/3987683/homebrew-install-specific-version-of-formula/9832084#9832084 so we perhaps want to use that. Or disable brew install entirely.

Reply:
its already an issue for CJXD too btw which took the same approach so I followed suit. 

how about we raise a separate issue of how to try get `jx upgrade cli` to work with specific versions and brew? its probably a thorny issue to get right while we figure out how to get the brew formula upgraded to a given jx version after we merge a version stream (with a timing issue if a user uses a new version stream before we update the brew formula). I guess we can just fail if the brew formula has not been updated yet?

Reply:
CJXD doesn't use brew at all (the instructions are to download from the website) so this is not an issue for CJXD.

Reply:
I've raised this issue to track upgrading `jx` to a version via brew https://github.com/jenkins-x/jx/issues/5290

Reply:
wonder if we should just ditch brew ;)